### PR TITLE
Add Powershell as shell

### PR DESCRIPTION
**Pull request by MattKotsenas**
- Add Powershell as a supported shell
  - There may be additional functionality to implement, but I've been
    running this way for a month with no known issues
- Update _get_shell to work with Windows
  - _get_shell assumed the parent process would always be the shell
    process, in Powershell the parent process is Python, with the
    grandparent being the shell
  - Switched to walking the process tree so the same code path can be used
    in both places
- Change Ctrl+C on Windows to match Unix

### CODE REVIEW

**Comment by scorphus** [comment]
Good job, @MattKotsenas. From what I can tell it looks okay, although the two logically distinct changes could be split into two independent commits.

I have a Windows box at home and will give it a shot soon tonight.

Thanks!

**Comment by MattKotsenas** [comment]
I've rebased to split the commit apart. Please let me know how it goes!

File: thefuck/shells/powershell.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from .generic import Generic
+
+class Powershell(Generic):
```
Comment (scorphus) [inline thefuck/shells/powershell.py]: It's a good style we follow to have 2 lines before a class definition.

File: thefuck/shells/powershell.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def app_alias(self, fuck):
+        return  'function ' + fuck + ' { \n' \
```
Comment (scorphus) [inline thefuck/shells/powershell.py]: Also, there are 2 spaces between `return` and `'`. Please remove that one and align it linedown.

File: tests/shells/test_powershell.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+import os
```
Comment (scorphus) [inline tests/shells/test_powershell.py]: This import isn't used. Can you please remove it?

File: thefuck/shells/__init__.py [hunk 4]
```diff
@@ -9,20 +9,37 @@
...
+
+        if name in shells:
+            return shells[name]()
+
+        try:
+            proc = proc.parent()
+        except TypeError:
+            proc = proc.parent
+
+    return Generic()
```
Comment (scorphus) [inline thefuck/shells/__init__.py]: Here you're creating an instance of `Generic`. `_get_shell` should return the class itself.

**Comment by scorphus** [comment]
@MattKotsenas You did great with the commit split! Thanks! Can you please fix that minor issues? I'm glad explaining any of them.

Also, this is how my `$profile` reads:
```powershell
# function fuck {
#     $fuck = $(thefuck (get-history -count 1).commandline)
#     if ("$fuck".startswith("echo")) {
#         "$fuck".substring(5)
#     } else {
#         if ($fuck) {
#             iex "$fuck"
#         }
#     }
# }

iex "thefuck --alias"
```
But this is what I got:
```
(py27) PS C:\Users\IEUser> fuck
Seems like fuck alias isn't configured!
Please put iex "thefuck --alias" in your $profile.
More details - https://github.com/nvbn/thefuck#manual-installation
(py27) PS C:\Users\IEUser>
```

What am I missing? I'm on a Windows 8.1 by the way.

Thanks.

**Comment by MattKotsenas** [comment]
Thanks @scorphus for the feedback. Stylistic feedback addressed and squashed into previous commit.

For the alias I'm not sure what's going on. If I do
```
iex "thefuck --alias" >> $profile
```

And restart PowerShell (necessary to pick up changes) I get the expected behavior. Am I interpreting the prompt wrong? Thanks!

**Comment by nvbn** [comment]
Thanks!

**Pull request merged**